trackers_integration/issuetracker/mantis.py

```diff
@@ -0,0 +1,140 @@
+
+    def add_comment(self, issue_id, body):
+        url = f"{self.base_url}{issue_id}/notes"
+        return self._request("POST", url, headers=self.headers, json=body)
+
+    def delete_comment(self, issue_id, note_id):
+        headers = {"Content-type": "application/json"}
+        url = f"{self.base_url}{issue_id}/notes/{note_id}"
+        return requests.request("DELETE", url, headers=headers, timeout=30)
+
+    @staticmethod
+    def _request(method, url, **kwargs):
+        return requests.request(method, url, timeout=30, **kwargs).json()
+
+
+class MantisThread(IntegrationThread):
+    """
+    Execute Mantis code in a thread!
+
+    Executed from the IssueTracker interface methods.
+
+    :meta private:
+    """
+
+    def post_comment(self):
+        comment_body = {
+            "text": markdown2html(self.text()),
+        }
+        self.rpc.add_comment(self.bug_id, comment_body)
+
+
+class Mantis(IssueTrackerType):
+    """
+    Support for Mantis. Requires:
+
+    :base_url: URL to Mantis Server - e.g. http://mantisbt
+    :api_url: Name of the Project in Mantis
+    :api_password: Mantis API token
+
+    .. note::
+        It is required to set ``api_url`` as Mantis Project name.
+        If not defined Kiwi will redirect to "Report Issue" page of Mantis.
+
+        You can leave the ``api_username`` field blank since
+        those are not used by issue tracker!
+    """
+
+    it_class = MantisThread
+
+    def _rpc_connection(self):
+        return MantisAPI(
+            self.bug_system.base_url,
+            self.bug_system.api_password,
+            self.bug_system.api_url,
+        )
+
+    def is_adding_testcase_to_issue_disabled(self):
+        return not (self.bug_system.base_url and self.bug_system.api_password)
+
+    def _report_issue(self, execution, user):
+        """
+        Mantis creates the Work Item with Title
+        """
+
+        create_body = {
+            "summary": f"Failed test: {execution.case.summary}",
+            "description": markdown2html(self._report_comment(execution, user)),
+            "category": {"name": "General"},
+            "project": {"name": self.bug_system.api_url},
+        }
+
+        try:
+            issue = self.rpc.create_issue(create_body)["issue"]
+
+            issue_url = f"{self.bug_system.base_url}/view.php?id={issue['id']}"
+            # add a link reference that will be shown in the UI
+            LinkReference.objects.get_or_create(
+                execution=execution,
+                url=issue_url,
+                is_defect=True,
+            )
+
+            return (issue, issue_url)
+        except Exception:  # pylint: disable=broad-except
+            # something above didn't work so return a link for manually
+            # entering issue details with info pre-filled
+            url = self.bug_system.base_url
+            if not url.endswith("/"):
+                url += "/"
+
+            return (None, url + "bug_report_page.php")
```

Comment:
Q: does this page support query string parameters so we can pre-fill some of the values automatically ?